cpp/src/rolling/rolling.cu

```diff
@@ -1283,22 +1287,67 @@ size_t multiplication_factor(cudf::data_type const& data_type)
   }
 }
 
-// Time-range window computation, with
-//   1. no grouping keys specified
-//   2. timetamps in ASCENDING order.
-// Treat as one single group.
-template <typename TimestampImpl_t>
+/// Given a single, ungrouped timestamp column, return the indices corresponding
+/// to the first null timestamp, and (one past) the last null timestamp.
+/// The input column is sorted, with all null values clustered either
+/// at the beginning of the column or at the end.
+/// If no null values are founds, null_begin and null_end are 0.
+std::tuple<size_type, size_type> get_null_bounds_for_timestamp_column(
+  column_view const& timestamp_column)
+{
+  auto num_rows  = timestamp_column.size();
+  auto num_nulls = timestamp_column.null_count();
+
+  if (num_nulls == num_rows) {
+    // Short-circuit: All nulls.
+    return std::make_tuple(0, num_rows);
+  }
+
+  auto first_row_is_null = timestamp_column.null_count(0, 1) == 1;
+  auto last_row_is_null  = timestamp_column.null_count(num_rows - 1, num_rows) == 1;
+
+  if (!timestamp_column.nullable() || (!first_row_is_null && !last_row_is_null)) {
+    // Neither first nor last rows are null. No nulls here.
+    return std::make_tuple(0, 0);
+  } else if (first_row_is_null) {
+    return std::make_tuple(0, num_nulls);
+  } else {
+    return std::make_tuple(num_rows - num_nulls, num_rows);
+  }
+}
```

Comment:
Just a suggestion since the flow is already pretty clear. The benefit is that there's one less branch, and one less kernel launch.
```suggestion
  auto const num_rows  = timestamp_column.size();
  auto const num_nulls = timestamp_column.null_count();

  if (num_nulls == num_rows || num_nulls == 0) {
    // Short-circuit: All nulls or no nulls.
    return std::make_tuple(0, num_nulls);
  }

  auto const last_row_is_null  = timestamp_column.null_count(num_rows - 1, num_rows) == 1;
  if (last_row_is_null)) {
    // nulls at the end 
    return std::make_tuple(num_rows - num_nulls, num_rows);
  } else {
    // nulls at the beginning
    return std::make_tuple(0, num_nulls);
  }
}
```

Reply:
Agreed. That would be cleaner.

Reply:
Argh, I think I messed up this change (from testing locally). I'll post a fix shortly.

Reply:
Fixed. Apologies for the churn.